### ISSUE TITLE
Make fancy filter work with user table on Request page

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#3016]
<sup>ℹ️ [Figma (root)][figma] | [Sitemap][sitemap]</sup>
<sup>_This initial comment is collaborative and open to modification by all._</sup>

## Description

If a PoolCandidateSearchRequest contains an applicantFilter instead of a poolCandidateFilter, the request page must use the 2.0 version of the User table, with extra filters and updated style. It must also use the same same search query as the talent-search page 2.0.

## Screenshot: Prototype
[![a screenshot of ...][image]][image-link]
<sup>[Click screenshot to visit Figma.][image-link]</sup>

## Acceptance Criteria
- [ ] If the request has an applicantFilter, the page shows the same users which would have been returned on the Talent Search 2.0 page
    - [ ] Links to the Admin User page with the tabs
    - [ ] Has the nice filters from User Table 2.0
    - [ ] Allows for downloading of user profiles.
- [ ] If a request has a poolCandidateFilter and no applicantFilter, it can be unchanged.

   [image]: https://user-images.githubusercontent.com/4313717/174098348-c22a9852-c598-4ea0-be9f-6837a10df666.png
   [image-link]: https://www.figma.com/proto/XS4Ag6GWcgdq2dBlLzBkay/Admin-v3-(Profile-release)?node-id=410%3A1079&starting-point-node-id=410%3A1984
   [figma]: https://go.talent.c4nada.ca/figma
   [sitemap]: https://go.talent.c4nada.ca/sitemap

**Comment by mnigh** [comment]
An estimate of a 2 seems very low for this issue considering this will require re-creating the Users filters which seemed quite complicated and involved many parts. Since this issue was estimated months ago, has the scope changed?

**Comment by vd1992** [comment]
@mnigh 
I feel this is blocked or has a chunk split out into #3648

**Comment by patcon** [comment]
Note that the UI feature of adding a count is not yet implemented, and can prob be its own unit of work (some outstanding questions for design)

New ticket: https://github.com/GCTC-NTGC/gc-digital-talent/issues/3964

**Comment by mnigh** [comment]
> I feel this is blocked or has a chunk split out into #3648

@vd1992 Can you add the issues that this issue is blocked by to this issue? As you can probably tell by my previous comment, I am still a bit unsure on this issue's scope and dependencies.

**Comment by brindasasi** [comment]
Team to discuss the strategy to implement the filter

**Comment by tristan-orourke** [comment]
Closing as a duplicate of #3648
